internal/bundlereader/auth_test.go

```diff
@@ -0,0 +1,148 @@
+package bundlereader_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/rancher/fleet/internal/bundlereader"
+	"github.com/rancher/fleet/internal/mocks"
+)
+
+// nolint: funlen
+func TestReadHelmAuthFromSecret(t *testing.T) {
+	cases := []struct {
+		name              string
+		secretData        map[string][]byte
+		getError          string
```

Comment:
This field is always empty. Did you mean to add a test case for secret retrieval errors?

Reply:
:) yup, I've added the missing test case